src/engine/renderer/tr_shade_calc.cpp

```diff
@@ -435,6 +435,27 @@ float RB_EvalExpression( const expression_t *exp, float defaultValue )
 	return GetOpValue( &ops[ 0 ] );
 }
 
+float RB_EvalExpression( const expression_t *exp, float defaultValue )
+{
+	ASSERT( exp );
+
+	float value = EvalExpression( exp, defaultValue );
+
+	if ( exp->bits & EXP_NORM )
+	{
+		value = Math::Clamp( value, 0.0f, 1.0f );
+	}
+
+	if ( exp->bits & EXP_SRGB )
```

Comment:
I don't think it makes sense to do this here since there are probably other cgen's where you want to convert `tess.svars.color`:
- `CGEN_CONST`
- `CGEN_WAVEFORM`
- `CGEN_ENTITY`
- `CGEN_ONE_MINUS_ENTITY`

Reply:
Hmmmm, `CGEN_CONST` can be converted at load time when parsing the shader, now done.

Reply:
For example CG_DrawSphere and CG_RenderParticle.

Reply:
> Hmmmm, `CGEN_CONST` can be converted at load time when parsing the shader, now done.

Hmm, it would be bad if such shader is used in UI.

Reply:
Well, maybe such shader can't be used in UI to begin with… I don't know.

Reply:
OK, so this now converts `CGEN_CONST` at run time and only if not in 2D backend. I modified some other conversion to also take care of the render being UI or not.